src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -191,4 +198,48 @@
                     return Response.ok(json).build();
                 });
     }
+
+    @POST
+    @Path("/email/subscription")
+    @RolesAllowed("read")
+    public Uni<Response> updateEmailSubscription(@Context SecurityContext sec, @NotNull @Valid SubscriptionSettings subscriptionSettings) {
+        RhIdPrincipal principal = (RhIdPrincipal) sec.getUserPrincipal();
+
+        Uni<Boolean> instantEmail;
+        Uni<Boolean> dailyEmail;
+
+        if (subscriptionSettings.isInstantEmail()) {
+            instantEmail = emailSubscriptionResources.subscribe(
+                    principal.getAccount(),
+                    principal.getName(),
+                    EmailSubscriptionType.INSTANT
+            );
+        } else {
+            instantEmail = emailSubscriptionResources.unsubscribe(
+                    principal.getAccount(),
+                    principal.getName(),
+                    EmailSubscriptionType.INSTANT
+            );
+        }
+
+        if (subscriptionSettings.isDailyEmail()) {
+            dailyEmail = emailSubscriptionResources.subscribe(
+                    principal.getAccount(),
+                    principal.getName(),
+                    EmailSubscriptionType.DAILY
+            );
+        } else {
+            dailyEmail = emailSubscriptionResources.unsubscribe(
+                    principal.getAccount(),
+                    principal.getName(),
+                    EmailSubscriptionType.DAILY
+            );
+        }
+
+        return Multi.createBy().merging()
+                .streams(instantEmail.toMulti(), dailyEmail.toMulti())
+                .collectItems().asList()
+                .onItem().transform(changedList -> Response.ok().build())
```

Comment:
If you're not interested in the items, you could use ``.onItem().ignoreAsUni()`` also.

Reply:
However, in this case, there's a better way. There's no need for Multi at all. So:

```java
Uni.combine().all().unis(instantEmail, dailyMail).discardItems()
```

For example would do the work. EndpointProcessor has this type of processing.